src/scenic/simulators/robosuite/simulator.py

```diff
@@ -0,0 +1,566 @@
+        
+        # Step simulation
+        for _ in range(self.physics_steps):
+            obs, reward, done, info = self.robosuite_env.step(action)
+            self._current_obs = obs
+            self._done = done
+            if self.render:
+                self.robosuite_env.render()
+            if done:
+                break
+        
+        self.pending_robot_action = None
+    
+    def getProperties(self, obj, properties: List[str]) -> Dict[str, Any]:
+        """Get current property values for an object."""
+        values = {}
+        robot_idx = self.robots.index(obj) if obj in self.robots else None
+        
+        for prop in properties:
+            if prop == 'position':
+                values[prop] = self._get_object_position(obj)
+            elif prop == 'joint_positions' and robot_idx is not None:
+                values[prop] = self._get_robot_joints(robot_idx)
+            elif prop == 'eef_pos' and robot_idx is not None:
+                values[prop] = self._get_eef_position(robot_idx)
+            elif prop == 'gripper_state' and robot_idx is not None:
+                values[prop] = self._get_gripper_state(robot_idx)
+            else:
+                values[prop] = getattr(obj, prop, None)
+        
+        return values
+    
+    def _get_object_position(self, obj) -> Vector:
+        """Get object position."""
+        obj_name = self.object_name_map.get(obj)
+        if obj_name and self._current_obs:
+            obs_key = f"{obj_name}_pos"
+            if obs_key in self._current_obs:
+                pos = self._current_obs[obs_key]
+                return Vector(pos[0], pos[1], pos[2])
+            elif obj_name in self.body_id_map:
+                body_id = self.body_id_map[obj_name]
+                pos = self.data.xpos[body_id]
+                return Vector(pos[0], pos[1], pos[2])
+        return obj.position
+    
+    def _get_robot_joints(self, robot_idx: int) -> List:
+        """Get robot joint positions."""
+        if robot_idx < len(self.robosuite_env.robots):
+            return list(self.robosuite_env.robots[robot_idx]._joint_positions)
+        return []
+    
+    def _get_eef_position(self, robot_idx: int) -> List:
+        """Get end-effector position."""
+        if self._current_obs:
+            return list(self._current_obs.get(f'robot{robot_idx}_eef_pos', [0, 0, 0]))
+        return [0, 0, 0]
+    
+    def _get_gripper_state(self, robot_idx: int) -> List:
+        """Get gripper state."""
+        if self._current_obs:
+            return list(self._current_obs.get(f'robot{robot_idx}_gripper_qpos', [0, 0]))
+        return [0, 0]
+    
+    def getCurrentObservation(self) -> Optional[Dict]:
+        """Get current observation dictionary."""
+        return self._current_obs
+    
+    def checkSuccess(self) -> bool:
+        """Check if task is successfully completed."""
+        if hasattr(self.robosuite_env, '_check_success'):
```

Comment:
How can this condition ever be false? More broadly, what's the point of this method?